src/editor.js

```diff
@@ -104,6 +115,58 @@ CKEDITOR.define( [
 					};
 				}
 			}
+
+			function prepareData() {
+				// At this stage we have the editor initial data at its "pure" form. There are two possibilities:
+				//
+				//   1. editor.setData() has been called before editor.init() - editor._data has been set. In this case
+				//      we don't want to change it because we assume that this is the data the end-developer wants to
+				//      have, no matter what.
+				//
+				//   2. An editable is available. In this case we take the initial data from editable.element. This must
+				//      be done at this stage because both fireCreator() and initEditables() will drive the data to
+				//      editable.view instead.
+				if ( typeof that._data != 'string' ) {
+					that._data = that.getData();
+				}
+			}
+
+			function fireCreator() {
+				// Take the name of the creator to use (config or any of the registered ones).
+				var creatorName = config.creator || Object.keys( that._creators )[ 0 ];
```

Comment:
It would be better if creators would have `match` method and the first matching creator would be used. It means I can add classic creator for `textarea` element, inline for `div` and oneline for `input` and proper creator would be used if the `create` method will be called on the element.